ananse/peakpredictor.py

```diff
@@ -0,0 +1,617 @@
+                self._motifs[factor] = motif_df[self.f2m[factor]].mean(1)
+
+    def _load_reference_data(self):
+        """Load data for reference regions.
+
+        Will load three types of data:
+        * Motif scores.
+        * The average peak coverage.
+        * The distance from the peak to nearest TSS.
+        """
+        # Read motifs
+        logger.info("loading motifs for reference")
+        self._motifs = pd.read_feather(f"{self.data_dir}/reference.factor.feather")
+        self._motifs.set_index(self._motifs.columns[0], inplace=True)
+
+        # Read average coverage
+        logger.info("loading average peak coverage for reference")
+        self._avg = pd.read_table(
+            f"{self.data_dir}/reference.coverage.txt",
+            sep="\t",
+            comment="#",
+            index_col=0,
+        )
+        self._avg.columns = ["average"]
+        self._avg["average"] = self._avg["average"] / self._avg["average"].max()
+
+        # Read distance to TSS
+        logger.info("loading distance for reference")
+        self._dist = pd.read_table(
+            f"{self.data_dir}/reference.dist_to_tss.txt",
+            sep="\t",
+            comment="#",
+            index_col=0,
+        )
+
+        # Set regions
+        self.regions = self._avg.index
+
+    def _load_human_factors(self):
+        valid_factors = pd.read_excel(
+            "https://www.biorxiv.org/content/biorxiv/early/2020/12/07/2020.10.28.359232/DC1/embed/media-1.xlsx",
+            engine="openpyxl",
+            sheet_name=1,
+        )
+        valid_factors = valid_factors.loc[
+            valid_factors["Pseudogene"].isnull(), "HGNC approved gene symbol"
+        ].values
+        valid_factors = [f for f in valid_factors if f not in ["EP300"]]
+        return valid_factors
+
+    def is_human_genome(self):
+        base_genome = os.path.basename(self.genome)
+        for name in ["hg38", "GRCh38", "hg19", "GRCh37"]:
+            if name in base_genome:
+                return True
+
+    def factors(self):
+        if self.is_human_genome():
+            valid_factors = self._load_human_factors()
+            return [f for f in self.f2m if f in valid_factors]
+        return list(self.f2m.keys())
+
+    def _load_factor2motifs(self, pfmfile=None, indirect=True, factors=None):
+        motifs = read_motifs(pfmfile, as_dict=True)
+        f2m = {}
+
+        if self.is_human_genome():
+            valid_factors = self._load_human_factors()
+
+        for name, motif in motifs.items():
+            for factor in get_motif_factors(motif, indirect=indirect):
+                if factors is not None and factor not in factors:
+                    continue
+
+                # TODO: this is temporary, while the motif database we use
+                # not very clean...
+                if self.is_human_genome():
+                    factor = factor.upper()
+
+                if self.is_human_genome() and factor not in valid_factors:
```

Comment:
are we sure that **all** valid_factors are actually uppercase in the case of human?

Reply:
Should be!